modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache

```diff
@@ -104,37 +102,46 @@ export class BaseAPI {
         return this.withMiddleware<T>(...middlewares);
     }
 
-    protected async request(context: RequestOpts, initOverrides?: RequestInit): Promise<Response> {
-        const { url, init } = this.createFetchParams(context, initOverrides);
+    protected async request(context: RequestOpts, initOverrides?: RequestInit | ((requestContext: { init: HTTPRequestInit, context: RequestOpts }) => Promise<RequestInit>)): Promise<Response> {
+        const { url, init } = await this.createFetchParams(context, initOverrides);
         const response = await this.fetchApi(url, init);
         if (response.status >= 200 && response.status < 300) {
             return response;
         }
         throw new ResponseError(response, 'Response returned an error code');
     }
 
-    private createFetchParams(context: RequestOpts, initOverrides?: RequestInit) {
+    private async createFetchParams(context: RequestOpts, initOverrides?: RequestInit | ((requestContext: { init: HTTPRequestInit, context: RequestOpts }) => Promise<RequestInit>)) {
         let url = this.configuration.basePath + context.path;
         if (context.query !== undefined && Object.keys(context.query).length !== 0) {
             // only add the querystring to the URL if there are query parameters.
             // this is done to avoid urls ending with a "?" character which buggy webservers
             // do not handle correctly sometimes.
             url += '?' + this.configuration.queryParamsStringify(context.query);
         }
-        const body = ((typeof FormData !== "undefined" && context.body instanceof FormData) || context.body instanceof URLSearchParams || isBlob(context.body))
-        ? context.body
-        : JSON.stringify(context.body);
 
         const headers = Object.assign({}, this.configuration.headers, context.headers);
         Object.keys(headers).forEach(key => headers[key] === undefined ? delete headers[key] : {});
 
-        const init = {
-            method: context.method,
-            headers: headers,
-            body,
-            credentials: this.configuration.credentials,
-            ...initOverrides
-        };
+        const initOverrideFn: (requestContext: { init: HTTPRequestInit, context: RequestOpts }) => Promise<RequestInit> = typeof initOverrides === "function" ? initOverrides : async (requestContext: { init: HTTPRequestInit, context: RequestOpts }) => ({...requestContext.init, ...initOverrides});
```

Comment:
I suggest to create a type alias of 
requestContext: { init: HTTPRequestInit, context: RequestOpts }) => Promise<RequestInit>

And use that everywhere

Reply:
@macjohnny Thank you for your review :)

I fixed this at 89d73f6 !